src/test/java/org/minions/devfund/bruno/WhackAMoleTest.java

```diff
@@ -0,0 +1,232 @@
+    @Test
+    public void validTheTwoCoordinates() {
+        final int xWhackMole = 4;
+        final int yWhackMole = 5;
+        assertTrue(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the coordinates are are inside the size grid.
+     */
+    @Test
+    public void inValidTheTwoCoordinates() {
+        final int xWhackMole = -1;
+        final int yWhackMole = 11;
+        assertFalse(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the coordinates are are inside the size grid.
+     */
+    @Test
+    public void inValidTheXCoordinates() {
+        final int xWhackMole = -1;
+        final int yWhackMole = 8;
+        assertFalse(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the coordinates are are inside the size grid.
+     */
+    @Test
+    public void inValidTheYCoordinates() {
+        final int xWhackMole = 4;
+        final int yWhackMole = 11;
+        assertFalse(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the coordinates are are inside the size grid.
+     */
+    @Test
+    public void inValidTheXExceedCoordinates() {
+        final int xWhackMole = 15;
+        final int yWhackMole = -1;
+        assertFalse(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the coordinates are are inside the size grid.
+     */
+    @Test
+    public void inValidTheXYCoordinates() {
+        final int xWhackMole = -15;
+        final int yWhackMole = -1;
+        assertFalse(game.validCordinates(xWhackMole, yWhackMole));
+    }
+
+    /**
+     * Verifies if the game should continue.
+     */
+    @Test
+    public void gameOverForRendition() {
+        final int xWhackMole = -1;
+        final int yWhackMole = -1;
+        game.placeAllMoles(NUM_MOLES);
+        game.whack(xWhackMole, yWhackMole);
+        assertTrue(game.gameOver());
+    }
+
+    /**
+     * Verifies if the game should continue.
+     */
+    @Test
+    public void gameOverContinuePlaying() {
+        final int xWhackMole = 7;
+        final int yWhackMole = 8;
+        game.placeAllMoles(NUM_MOLES);
+        game.whack(xWhackMole, yWhackMole);
+        assertFalse(game.gameOver());
+    }
+
+    /**
+     * Verifies if the game should continue.
+     */
+    @Test
+    public void gameOverWithoutAttempts() {
+        game.placeAllMoles(NUM_MOLES);
+        game.setAttemptsLeft(0);
+        assertTrue(game.gameOver());
+    }
+
+    /**
+     * Verifies if the game should continue.
+     */
+    @Test
+    public void gameOverWin() {
+        game.setMolesLeft(0);
+        assertTrue(game.gameOver());
+    }
+
+    /**
+     * Verifies if the method generate a grind to print.
+     */
+    @Test
+    public void printGridToUser() {
+        game.printGridToUser();
+    }
+
+    /**
+     * Verifies if the exit game should be false.
+     */
+    @Test
+    public void exitGameFalse() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "exitGameShouldBeFalse" on line 106. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)